homework/nwd-nnw/nwdNww.hpp

```diff
@@ -1,11 +1,39 @@
 #pragma once
 
-int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+int abs(int a){
+    if (a >= 0) return a;
+     else return a*(-1);
 }
 
+
+    int NWD(int lhs, int rhs) {
+        int myNWD;
+        lhs = abs(lhs);
+        rhs=abs(rhs);
+
+        if (lhs == 0)
+            myNWD= rhs;
+        if (rhs == 0)
+            myNWD= lhs;
```

Comment:
If `lhs == 0` or `rhs == 0`, then you still continue to the ` while( lhs != rhs )` loop because the condition may be true, and your code has an infinite loop